[client] Issue 리스트 구성 컴포넌트 설정

ahrimy opened the pull request:
<img width="557" alt="Screen Shot 2020-11-03 at 6 58 11 PM" src="https://user-images.githubusercontent.com/26592306/97973962-44572480-1e0a-11eb-91f4-d9c74e26042d.png">

- `React.StrictMode` 적용
- `pages` : 각각의 페이지 
- `components` : 각각의 페이지를 구성하는 컴포넌트
- Header component 생성 및 App 컴포넌트에 추가
- <del>`IssueListPage` : `SearchBar`, `Button`, `IssueList (IssueListHeader + IssueItem)`컴포넌트 들로 구성</del>

**컴포넌트 구조수정**
```html
<App>
  <Header></Header>
  <IssueListPage>
    <IssueListNav>
      <Searchbar>
        <filter></filter>
        <textbox></textbox>
      </Searchbar>
      <btns>  <!--버튼은 아직 각각 컴포넌트는 생성 안함-->
        <labelBtn></labelBtn>
        <milestoneBtn></milestoneBtn>
        <newIssueBtn></newIssueBtn>
      </btns>
    </IssueListNav>
   <IssueList>
     <IssueListHeader>
       ...filters...
     </IssueListHeader>
     <IssueListBody>
       <IssueItem></IssueItem>
       ...
       <IssueItem></IssueItem>
     </IssueListBody>
   </IssueList>
  </IssueListPage>
</app>
```

구조를 어느정도까지 구현을 해놔야 할지 모르겠어서 우선 IssueListPage만 추가해봤습니다. 리뷰 해주시면 그 다음부분 진행방향 잡을게요!

Commit:
[client] chore: React StrictMode 설정

- 개발 과정에서 React 애플리케이션 오류를 확인을 위한 StrictMode 코드 추가

Commit:
[client] feat: 헤더 컴포넌트 추가

- header component 파일 생성
- header component를 app component 에 추가

Commit:
[client] chore: component 폴더 이름 수정

- component 폴더를 components로 이름 변경

Commit:
[client] feat: Issue 리스트 구성 컴포넌트 생성

- Issue 리스트 페이지 생성
- Issue 리스트 페이지 안의 컴포넌트 생성
- SearchBar : Issue 검색 창
- IssueList: Issue 리스트 전체
- IssueItem: Issue 리스트의 각 항목 컴포넌트
- IssueListHeader: Issue 리스트 상단의 필터 컴포넌트

skid901 commented on web/client/src/pages/IssueListPage.js:
```diff
@@ -0,0 +1,16 @@
+      <button>New Issue</button>
+      <br /> <br />
+      <IssueList />
+    </>
```
`<React.Fragment />` 대신 `<div />` 를 통해서, SearchBar 와 그 우측에 위치한 버튼들을 한번 묶어주면 좋을 것 같아요.

`<React.Fragment />` 를 사용하게 되면 서치바와 버튼들을 배치하기 위한 영역을 잡아주는 태그가 없어서 css 입히기가 힘들 것 같습니다.

[React.Fragment 참고링크](https://ko.reactjs.org/docs/fragments.html)를 한번 참고해 보시면 좋을 것 같아요.

ahrimy replied:
css 적용하는걸 생각을 못했네용.....이부분 말고도 Fragment 사용한 부분에 <div /> 넣어야하는 거 다시 한번 확인해볼게요!

seokju2ng replied:
경덕님 말씀 들어보니 `<></>` 부분이 `React.Fragment`인 것 같아요!
컴포넌트 단위로 분리할 때 예시링크의 테이블 내부를 분리한다거나 하는 부분 아니면 대부분 `div`로 묶여야할 것 같아요. 혹시 시멘틱 태그가 가능하면 시멘틱 태그로 쓰거나..?

ahrimy replied:
이 부분 전체적으로 수정했습니당

skid901 commented on web/client/src/App.js:
```diff
@@ -1,6 +1,13 @@
 import React from 'react';
```
`import React from 'react';` 구문을 생략해도, 웹팩에서 리액트 린트를 통해 자동으로 삽입해 주기 때문에, 직접 기재해주지 않아도 되는 걸로 알고 있습니다.

혹시 생략해도 코드가 정상적으로 작동되면 생략하는 걸 컨벤션으로 정해도 좋을 것 같은데, 두 분은 어떻게 생각하시나요?

@seokju2ng @ahrimy

seokju2ng replied:
실제 리액트 컨벤션은 어떻게 되는지 알고 계실까요? 모르셔도 리액트 접해본 두 분이 정해주시는 것으로 가면 좋을 것 같아요!

ahrimy replied:
import 해야지 사용할 수 있는줄 알았는데 안해도 작동이 된다면 굳이 기재하지않아도 될 것 같네요!

skid901 replied:
직접 생략해서 돌려봤는데 지금은 에러가 발생하네요;;
다시 확인해보고 어떻게 하면 생략해도 되는지 확인해 보겠습니다 ^^;;

ahrimy replied:
그럼 이부분은 우선은 import하는 걸로 해두겠습니다!

seokju2ng reviewed (commented):
코드만 보고 대충 이해한 내용으로 리뷰 드리면,
컴포넌트 단위를 제가 생각한 내용이랑 좀 다르게 잡으셔서 제 생각을 말씀드릴게요. 어떠신지 댓글 부탁드려요!
제가 생각하는 컴포넌트 구조(?)는 대충 이렇습니다.
```html
<app>
  <header></header>
  <issueListPage>
    <nav>
      <searchbar>
        <filterBtn></filterBtn>
        <textbox></textbox>
      </searchbar>
      <btns>
        <labelBtn></labelBtn>
        <milestoneBtn></milestoneBtn>
        <newIssueBtn></newIssueBtn>
      </btns>
    </nav>
    <issueBody>
      <issueListHeader>
        ...filters...
      </issueListHeader>
      <issueList>
        <issueItem></issueItem>
        ...
        <issueItem></issueItem>
      </issueList>
    </issueBody>
  </issueListPage>
</app>
```

ahrimy reviewed (commented):
석중님이 보여주신 구조처럼 SearchBar랑 다른 Button들을 하나의 컴포넌트로 만드는 작업은 필요할것 같네요! 
 ``` html
    <issueList>
      <issueListHeader>
        ...filters...
      </issueListHeader>
      <issueListBody>
        <issueItem></issueItem>
        ...
        <issueItem></issueItem>
      </issueListBody>
    </issueList>
````
IssueListBody 부분을 이렇게 하는건 어떤가요??
@seokju2ng @skid901

seokju2ng commented:
구조는 제 것과 같은 데, 컴포넌트 네임을 말씀하시는 건가요????
네임이라면 저는 이름 정하는데 시간쓰기 싫어서 일단 설명드릴 요량으로 대충 적어서 네이밍은 전혀 괜찮습니다 ㅋㅋㅋㅋ 아림님이 원하시는 대로 잡아주세요!

skid901 commented:
> IssueListBody 부분을 이렇게 하는건 어떤가요??

@ahrimy  적어주신 대로 하는거 좋은 것 같아요!!

Commit:
[client] fix: 컴포넌트의 최상위에<div /> 태그 추가

- React.Fragment로 처리해주었던 각 컴포넌트의 최상위에 <div />태그 추가

Commit:
[client] feat: Issue List Page에 Nav 컴포넌트 추가

- Search Bar, Labels Button, Milestone Button, New Issue Button을 하나의 Nav 컴포넌트에 추가

Commit:
[client] feat: issue list에 body 컴포넌트 추가

- IssueListBody 컴포넌트 추가
- IssueItem을 IssueListBody컴포넌트 안에 추가

Commit:
[client] fix: IssueListNav의 button들의 상위에 <div />태그 추가

- IssueListNav의 버튼들을 한번에 관리하기 위해서 <div />태그 추가

Commit:
[client] fix: IssueListNav의 button들의 상위에 <div />태그 추가

- IssueListNav의 버튼들을 한번에 관리하기 위해서 <div />태그 추가

Commit:
[client] style: 컴포넌트 함수명 수정

- 각각의 컴포넌트의 함수명 수정

Commit:
Merge branch 'feat/client/issuelist-layout' of https://github.com/boostcamp-2020/IssueTracker-16 into feat/client/issuelist-layout

ahrimy commented:
push 하고나서 commit을 수정하려고 했더니 commit들이 약간 꼬여서 마지막에 좀 깔끔하지 않게 push 됐네용...죄송합니다ㅜ

skid901 reviewed (approved):
석중 님이 말씀해주신 `semantic tag`를 적용하는 걸 같이 해보면 좋을 거 같아요.

일단은 머지를 하고, 시멘틱 태그에 대해 같이 좀더 살펴보고 나서 `styled-component`를 사용할 때 시멘틱 태그를 다시 적용해보면 좋을거 같습니다!!

그 외에는 그림 그려주신 대로, 컴포넌트를 명확하게 나누어서 파일을 구성해 주신 것 같습니다.

오랜 시간 작업 고생하셨습니다.
